src/contract.rs

```diff
@@ -1,66 +1,122 @@
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
 
+    let mut seen_addresses: Vec<String> = vec![];
     let mut total_ownership: u8 = 0;
 
-    // Check that the sum of the individual ownerships is equal to 100
     for owner in &msg.owners {
         total_ownership += owner.ownership;
+
+        if seen_addresses.contains(&owner.address.to_string()) {
+            return Err(StdError::generic_err(
+                "Duplicate owner address has been input more than once",
+            ));
+        } else {
+            seen_addresses.push(owner.address.to_string());
+        }
     }
 
+    // Check if the total ownership is equal to 100%. If not, return an error.
     if total_ownership != 100 {
         return Err(StdError::generic_err("Total Ownership must equal 100%."));
     }
 
+    // Check if each owner's individual ownership is greater than 0%. If not, return an error.
     for owner in &msg.owners {
         if owner.ownership == 0 {
             return Err(StdError::generic_err(
-                "Individual Ownership must be greater than 0",
+                "Individual Ownership must be greater than 0.",
             ));
         }
     }
 
+    // Save the owners and their ownership percentages to storage.
     CONFIG.save(deps.storage, &msg.owners)?;
 
-    Ok(Response::default())
+    // Return a successful `Response`.
+    Ok(Response::new().add_attribute("action", "instantiate"))
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
+// Define the `execute` entry point function, which takes in several arguments and returns a `Result<Response, ContractError>`.
 pub fn execute(
     deps: DepsMut,
-    _env: Env,
+    env: Env,
     info: MessageInfo,
     msg: ExecuteMsg,
 ) -> Result<Response, ContractError> {
+    // Match on the message type received and execute the appropriate code based on the type.
     match msg {
-        ExecuteMsg::Disburse {} => disburse(deps, info),
+        ExecuteMsg::Disburse {} => disburse(deps, info, env),
     }
 }
 
-pub fn disburse(mut _deps: DepsMut, _info: MessageInfo) -> Result<Response, ContractError> {
-    todo!()
+// Define the `disburse` function, which takes in several arguments and returns a `Result<Response, ContractError>`.
+pub fn disburse(deps: DepsMut, info: MessageInfo, _envv: Env) -> Result<Response, ContractError> {
+    // Load the `config` data from storage.
+    let config = CONFIG.load(deps.storage)?;
 
-    // dusbursing of funds follows this logic:
+    // Check if the sender is authorized to disburse funds by iterating over the `config` and looking for a matching address.
+    let authorized = config.iter().any(|owner| owner.address == info.sender);
+    if !authorized {
+        return Err(StdError::generic_err("Unauthorized to disburse funds.").into());
+    }
 
+    // Build messages to disburse funds to each owner based on their ownership percentage.
+    let messages = build_messages(&info.funds, &config);
```

Comment:
`disburse` is a very small function, I wouldn't use another function, I would just do it in here.

Reply:
Ok, I can refactor to remove build_messages() and perform the logic in disburse() instead.